og_migrate/src/Plugin/migrate/OgEntityDeriver.php

```diff
@@ -0,0 +1,268 @@
+              }
+            }
+          }
+        }
+      }
+    }
+    catch (\Exception $e) {
+      // Return the remaining derivatives, if any.
+    }
+
+    return $this->derivatives;
+  }
+
+  /**
+   * Gets the migration process definition for an entity_type.
+   *
+   * @param string $group_type_migration
+   *   The group_type migration id.
+   * @param string $entity_type_migration
+   *   The entity_type migration id.
+   *
+   * @return array
+   *   The "process" definition.
+   */
+  protected function getEntityProcessDefinition($group_type_migration, $entity_type_migration) {
+    return [
+      'target_id' => [
+        [
+          'plugin' => 'migration_lookup',
+          'migration' => $group_type_migration,
+          'source' => 'gid',
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => 'Migration group is missing',
+        ],
+      ],
+      'entity_id' => [
+        [
+          'plugin' => 'migration_lookup',
+          'migration' => $entity_type_migration,
+          'source' => 'etid',
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => 'Migration entity is missing',
+        ],
+      ],
+      'entity_type' => 'entity_type',
+      'group_type' => 'group_type',
+      'field_name' => [
+        'plugin' => 'static_map',
+        'source' => 'field_name',
+        'map' => [
+          'og_group_ref' => 'og_audience',
+          'default' => 'og_audience',
+        ],
+      ],
+      'language' => 'language',
+    ];
+  }
+
+  /**
+   * Gets the migration process definition for user.
+   *
+   * @param string $group_type_migration
+   *   The group_type migration id.
+   *
+   * @return array
+   *   The "process" definition.
+   */
+  protected function getUserProcessDefinition($group_type_migration) {
+    return [
+      'id' => 'id',
+      'type' => [
+        [
+          'plugin' => 'migration_lookup',
+          'migration' => 'd7_og_membership_type',
+          'source' => 'type',
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => 'Membership type is missing',
+        ],
+      ],
+      'uid' => [
+        [
+          'plugin' => 'migration_lookup',
+          'migration' => 'd7_user',
+          'source' => 'etid',
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => 'User is missing',
+        ],
+      ],
+      'entity_id' => [
+        [
+          'plugin' => 'migration_lookup',
+          'migration' => $group_type_migration,
+          'source' => 'gid',
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => 'Migration group is missing',
+        ],
+      ],
+      'entity_type' => 'group_type',
```

Comment:
Found one problem that I introduced after my refactor: there was a reason why I made the entity_type lookup plugin. So that I could populate entity_bundle. :-)